### ISSUE TITLE
Receive notifications of new images uploaded to FathomNet in their area of expertise; user selects frequency and concepts of interest

### ISSUE THREAD

**Issue by hohonuuli** [fathomnet/community-feedback#54]
Related to bio information?

**Comment by kevinsbarnard** [comment]
UI component: user should be able to select frequency of notifications (daily, weekly, monthly, etc.)

**Comment by hohonuuli** [comment]
Related to #97 and #65

**Comment by hohonuuli** [comment]
Reminder that notifications need a mechanism to opt-in and opt-out. This requires changes to both the backend and frontend.

**Comment by ermbutler** [comment]
Same status - not started yet.

**Comment by hohonuuli** [comment]
Notifications are added in fathomnet-support

**Comment by hohonuuli** [comment]
The email will just pull content from their activity stream. Or just link to their activity stream.

**Comment by hohonuuli** [comment]
Just piggy back off of followed stream

**Comment by hohonuuli** [comment]
topics have a boolean notification field. Next step is to add two fields to FathomnetIdentity:

1. `enableNotifications: Boolean` - Allows a user to enable/disable notifications globally
2. `notificationFrequencey: Duration` - Allows the user to set how often to receive notifications.

**Comment by hohonuuli** [comment]
Starting work on <https://github.com/fathomnet/fathomnet-rest-api/tree/feature/54>

**Comment by hohonuuli** [comment]
Flow:

- A user subscribes to topics. They can enable/disable notification individually per topic. All subscribed topics show up in their activity stream
- A users enables/disables notifications. If enabled an email is sent at the notificationFrequency set in their account that summarized the activity.

**Comment by hohonuuli** [comment]
Added support for notification frequency in [v1.8.1](https://github.com/fathomnet/fathomnet-rest-api/releases/tag/1.8.1)

Next steps are to add a support service that sends email notifications of activity.

**Comment by hohonuuli** [comment]
@kevinsbarnard @ermbutler @kakanikatija When a user gets a notification email, what do we want it to say? Thoughts??

**Comment by kevinsbarnard** [comment]
@hohonuuli I'm thinking a summary of activity (based on what they follow) since the last email sent to them, and a link to the latest activity page so they can see the details. I'd say we should keep it simple for now, but I could see that email as a good place to put other semi-frequent updates in the future as well.

**Comment by ermbutler** [comment]
Would be good to run past Lilly -- I can ask her? I think we talked about it just being a link to keep things simple. We could do a summary too, or limit it to summary of a few updates and a "see all"

We should include links to opt out etc... Here is just a stab:
 
**Subject line:** New updates to your followed [concept/person/region]!

**Body:**
[Fathomnet logo]
**Heading:** 23 updates containing [concept name]
View them in your activity feed ->

**Footer:**
FathomNet is an open-source image database for understanding our ocean and its inhabitants.

Explore
Contact us
Blog
Follow us [Links]


You are receiving this email because of a subscription. To change frequency or opt out visit your account here: [link]

**Comment by hohonuuli** [comment]
@ermbutler Yes. That's a good idea, we should run it by Lilly. 

### tl;dr 
I like your template. I think we should tweak it so that it's a summary of all topics a user follows.

### All the details
Just little background about the way I've set things up in the database and the notification code ATM:

- A user can set a notification frequency which is one of `never`, `weekly`, `monthly` or `quarterly`
- They can follow various `topics`. Every topic has a `notification` field that can be `true` or `false`.
- A cron job will run for each notification frequency and find just the users for that frequency. The job will grab the appropriate duration activity stream and filter out items from the stream where the topic's notification is false. 
- We can summarize the activity stream. __This would be one email for all topics rather than individual emails for each topic (concept/person/region)__

## Revised template (straw man)

Subject line: FathomNet updates to your followed topics. 2023-05-31

Body:
[Fathomnet logo]
23 updates containing [concept name]
104 updates containing [a different concept name]
98 updates by [Followed Persons name]
1022 new images in [region name]
...

View them in your activity feed ->

Footer:
FathomNet is an open-source image database for understanding our ocean and its inhabitants.

Explore
Contact us
Blog
Follow us [Links]

You are receiving this email because of a subscription. To change frequency or opt out visit your account here: [link]

**Comment by hohonuuli** [comment]
Implemented in [ActivityNotifier](https://github.com/fathomnet/fathomnet-support/blob/main/src/main/scala/org/fathomnet/support/apps/notifiers/ActivityNotifier.scala). The current placeholder template is like:

```
Subject: FathomNet updates to your followed topics. 2023-05-31

FathomNet activity between 2023-05-24 and 2023-05-31:

 person

   9 updates to brian@mbari.org

 concept

   1825 updates to Nanomia bijuga
```

**Comment by kakanikatija** [comment]
I like Erin’s suggestions. It would be good to keep it simple to start (aka text only), and make it more fully formatted in the future.

**Comment by ermbutler** [comment]
@hohonuuli --- The only thing I would add back to that email is something like:
"You are receiving this email because of a subscription. To change frequency or opt out visit your account here: [link]"

Because of CAN-SPAM Act: A Compliance Guide for Business see number 5 here https://www.ftc.gov/business-guidance/resources/can-spam-act-compliance-guide-business 

And if we can add a location re: 4 in that list as well.
